app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala

```diff
@@ -81,8 +83,11 @@ class WhatIsYourEmailController @Inject()(actions: Actions, view: change_your_em
     emailForm.bindFromRequest.fold(
       formWithErrors => {
         subscriptionDisplayConnector.subscriptionDisplay(request.eori).map {
-          case SubscriptionDisplayResponse(Some(email)) => BadRequest(view(formWithErrors, email))
-        } recover { handleNonFatalException() }
+          case SubscriptionDisplayResponse(Some(email), None) => BadRequest(view(formWithErrors, email))
+          case SubscriptionDisplayResponse(None, Some(_)) => Ok(problemWithServiceView())
```

Comment:
could you add a `case SubscriptionDisplayResponse(None, None) => \\ToDo` for now, unless you are ok to cover that case in this story